### PR TITLE
[Examples] Add ASDKgram-swift to examples_extra

### PR DESCRIPTION
**Pull request by Cal-um**
This is a swift example of the Objective-C ASDKgram in the examples folder.

It is not an exact replica although does attempt to replicate the comparison between UIKIt and ASDK performance. The main differences are: 
- For the UIKit TableViewCell, autolayout is used exclusively 
- It does not show a comment view
- It does not demonstrate the visual style of flexbox layout
- Minor visual differences

It could prove beneficial to Swift who are unfamiliar with Objective-C like myself.

### CODE REVIEW

**Comment by hannahmbanana** [comment]
Woah - THANK YOU @Cal-um!

**Comment by Cal-um** [comment]
@hannahmbanana you are the original and best.

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableNodeController.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		node.view.leadingScreensForBatching = 2.5
+		navigationController?.hidesBarsOnSwipe = true
+
+	}
+
+	// helper functions
+
+	func configActivityIndicator() {
+
+		let activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .gray)
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableNodeController.swift]: Let's remove these blank lines

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableNodeController.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		self.navigationItem.title = "ASDK"
+
+	}
+
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	override func viewDidLoad() {
+		configActivityIndicator()
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableNodeController.swift]: Call `super.viewDidLoad()` here

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	}
+
+	override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+
+		let currentOffSetY = scrollView.contentOffset.y
+		let contentHeight = scrollView.contentSize.height
+		let screenHeight = UIScreen.main.bounds.size.height
+		let screenfullsBeforeBottom = (contentHeight - currentOffSetY) / screenHeight
+		if screenfullsBeforeBottom < 2.5 {
+				self.fetchNewBatch()
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableViewController.swift]: Nit: Extra indentation here

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/Webservice.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+			}
+		}.resume()
+	}
+}
+
+extension WebService {
+
+	// returns a tuple with .0 as error and .1 as data.
+
+	fileprivate func checkForNetworkErrors(_ data: Data?, _ response: URLResponse?, _ error: Error?) -> (NetworkingErrors?, Data?) {
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/Webservice.swift]: Let's have this return `Result<Data>`

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoTableNodeCell.swift [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		self.automaticallyManagesSubnodes = true
+	}
+
+	let photoImageNode: ASNetworkImageNode = {
+		let imageNode = ASNetworkImageNode()
+		imageNode.contentMode = .scaleAspectFill
+		return imageNode
+	}()
+
+	var avatarImageNode: ASNetworkImageNode = {
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoTableNodeCell.swift]: Use `let` here for consistency.

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoTableNodeCell.swift [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+	let photoLikesLabel: ASTextNode = {
+		let label = ASTextNode()
+		return label
+	}()
+
+	let photoDescriptionLabel: ASTextNode = {
+		let label = ASTextNode()
+		return label
+	}()
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoTableNodeCell.swift]: Let's move all the property declarations above `init` and trim these down to just

```swift
let usernameLabel = ASTextNode()
let timeIntervalLabel = ASTextNode()
etc.
```

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoModel.swift [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+			NSFontAttributeName: UIFont.systemFont(ofSize: size)
+		]
+
+		let date = Date.iso8601Formatter.date(from: self.dateString)!
+		return NSAttributedString(string: timeStringSince(fromConverted: date), attributes: attr)
+	}
+
+	private func timeStringSince(fromConverted date: Date) -> String {
+
+		let diffDates = NSCalendar.current.dateComponents(Set([.day, .hour, .second]), from: date, to: Date())
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoModel.swift]: Let's remove `Set()` here – the compiler will do the work for us.

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableViewController.swift [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+	}
+
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	override func viewDidLoad() {
+
+		activityIndicatorConfig()
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableViewController.swift]: Call `super.viewDidLoad()` here.

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableViewController.swift [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+	var screenSizeForWidth: CGSize = {
+		let screenRect = UIScreen.main.bounds
+		let screenScale = UIScreen.main.scale
+		return CGSize(width: screenRect.size.width * screenScale, height: screenRect.size.width * screenScale)
+	}()
+
+	// helper functions
+
+	func activityIndicatorConfig() {
+		let activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .gray)
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableViewController.swift]: Let's name this `setupActivityIndicator()`, and the method below `configureTableView()`

File: examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedModel.swift [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+//  FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+//   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+//  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import UIKit
+
+class PhotoFeedModel: NSObject {
```
Comment (Adlai-Holler) [inline examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedModel.swift]: Let's remove `: NSObject` here and make this `final class PhotoFeedModel`?

**Comment by Cal-um** [comment]
Hi Adlai-Holler, thanks so much for your review. I have made the changes. Let me know if there is anything else you need.

**Comment by Adlai-Holler** [comment]
Jenkins add to whitelist and test this please.

**Comment by Adlai-Holler** [comment]
OK @Cal-um sorry for the delay in merging this, and thanks for the diff! Here we go!

**Pull request merged**